codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java

```diff
@@ -40,13 +41,70 @@
  *
  * <p><em>This API is very immature.</em> The Http2Connection-based API is currently preferred over
  * this API. This API is targeted to eventually replace or reduce the need for the Http2Connection-based API.
+ *
+ * <h3>Opening and Closing Streams</h3>
+ *
+ * <p>When the remote side opens a new stream, the frame codec first emits a {@link Http2StreamActiveEvent} with the
+ * stream identifier set.
+ * <pre>
+ * Http2FrameCodec                                             Http2MultiplexCodec
+ *        +                                                             +
+ *        |         Http2StreamActiveEvent(streamId=3, headers=null)    |
+ *        +------------------------------------------------------------->
+ *        |                                                             |
+ *        |         Http2HeadersFrame(streamId=3)                       |
+ *        +------------------------------------------------------------->
+ *        |                                                             |
+ *        +                                                             +
+ * </pre>
+ *
+ * <p>When a stream is closed either due to a reset frame by the remote side, or due to both sides having sent frames
+ * with the END_STREAM flag, then the frame codec emits a {@link Http2StreamClosedEvent}.
+ * <pre>
+ * Http2FrameCodec                                         Http2MultiplexCodec
+ *        +                                                         +
+ *        |         Http2StreamClosedEvent(streamId=3)              |
+ *        +--------------------------------------------------------->
+ *        |                                                         |
+ *        +                                                         +
+ * </pre>
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Close this "DefaultHttp2FrameReader". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2095)